### PR TITLE
feature: unwrapkey operation for keyvault

### PR DESCRIPTION
**Pull request by reenigneEsrever92**
Just added the unwrapkey operation to azure_keyvault crate.

### CODE REVIEW

**Comment by reenigneEsrever92** [comment]
@microsoft-github-policy-service agree

**Review by heaths — APPROVED** [review]

File: sdk/security_keyvault/src/keys/operations/unwrap_key.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            let version = self.version.unwrap_or_default();
+            let mut uri = self.client.keyvault_client.vault_url.clone();
+            let path = format!("keys/{}/{}/unwrapkey", self.name, version);
+
+            uri.set_path(&path);
+
+            let mut request_body = Map::new();
+            request_body.insert(
+                "value".to_owned(),
+                Value::String(String::from_utf8(self.unwrap_key_parameters.ciphertext).unwrap()),
```
Comment (demoray) [inline sdk/security_keyvault/src/keys/operations/unwrap_key.rs:26]: A strong goal of the SDK is to actively avoid using `unwrap` in the SDK unless we have validated predicates that will ensure the SDK will not cause a panic.

I believe `azure_core::error::Error` has implementations to automatically convert `std::string::FromUtf8Error` (the error returned from `String::from_utf8`) to `azure_core::error::Error`.

Please replace this and the `unwrap` calls below with returning an error.

**Review by demoray — CHANGES_REQUESTED** [review]
Please replace the calls to `unwrap` with returning errors as indicated in my comment above.

File: sdk/security_keyvault/src/keys/operations/unwrap_key.rs (hunk 1 above)
Comment (reenigneEsrever92) [reply sdk/security_keyvault/src/keys/operations/unwrap_key.rs:26]: I wanted to do so. Yet, currently these operations are generated by a macro. Also I wanted the code to be inline with all the other operations and they do it that way.

Comment (demoray) [reply sdk/security_keyvault/src/keys/operations/unwrap_key.rs:26]: The `operation!` macro builds `UnwrapKey` to be the following:

```rust
pub type UnwrapKey = futures::future::BoxFuture<'static, azure_core::Result<UnwrapKeyResponse>>
```

Replacing `.unwrap()` calls in your PR with `?` handles converting the error types automatically.

**Review by demoray — CHANGES_REQUESTED** [review]

File: sdk/security_keyvault/src/keys/operations/unwrap_key.rs (hunk 1 above)
Comment (demoray) [reply sdk/security_keyvault/src/keys/operations/unwrap_key.rs:26]: Note, I just looked at the other operations in this crate.  The existing operations also use `unwrap`.  I'll be creating a PR to address that throughout the rest of this crate today.

Comment (demoray) [reply sdk/security_keyvault/src/keys/operations/unwrap_key.rs:26]: @reenigneEsrever92 I apologize for the confusion, as the existing code in this crate operated in the same fashion as your PR.

I've created https://github.com/Azure/azure-sdk-for-rust/pull/1312 to address this issue through the rest of this crate.  Hopefully the PR I linked demonstrates what is needed for this PR.

**Comment by reenigneEsrever92** [comment]
Thanks a lot! I've made changes accordingly.

**Review by demoray — APPROVED** [review]

**Comment by demoray** [comment]
Thanks for the contribution.

**Pull request merged**